src/main/java/seedu/address/model/person/ModuleName.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's module name in the address book.
+ */
+public class ModuleName {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Module name should consist of 2-3 Alphabets then 4 numbers. Optionally, there can be one more alphabet"
```

Comment:
Low priority but can make comment a bit clearer maybe
"The module name should follow this format:
Start with 2 or 3 uppercase letters.
Follow with exactly 4 digits.
Optionally, you may add one more letter after the digits.
Optionally, include a dash ('-') followed by one more letter at the end.
Example of a valid module name: CS1231S-T."

Reply:
i think displaying the example at the beginning first would make it easier to visualise the explanation of how ModuleName should be formatted